.github/workflows/main.yml

```
@@ -0,0 +1,164 @@
+name: Test, build and deploy
+
+on:
+  push: {}
+  pull_request: {}
+  schedule:
+    - cron: '0 0 * * *'
+
+env:
+  STREAMLINK_DIST_DIR: ${{ github.workspace }}/dist
+  SIGNING_KEY_ID: 2E390FA0
+  SIGNING_KEY_FILE: ${{ github.workspace }}/signing.key
+  DOCS_KEY_FILE: ${{ github.workspace }}/id_rsa_docs
+
+jobs:
+  test:
+    name: Test
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macOS-latest, windows-latest]
+        python: [2.7, 3.5, 3.6, 3.7, 3.8]
+#       include:
+#         - python: X.Y-dev
+#           continue: true
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 60
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python }}
+      - name: Install dependencies
+        continue-on-error: ${{ matrix.continue }}
+        run: bash ./script/install-dependencies.sh
+      - name: Test
+        continue-on-error: ${{ matrix.continue }}
+        run: pytest --cov
```

Can we include the `-ra` option for `pytest` to output some extra summary info at the end of the tests? It shows some extra details about why tests were skipped or why failures were allowed. 

```suggestion
        run: pytest -ra --cov
```

---

Done. Added this as another commit, as it's not been set previously.